boa/src/syntax/parser/statement/declaration/mod.rs

```diff
@@ -63,7 +63,7 @@ where
         let tok = cursor.peek(0)?.ok_or(ParseError::AbruptEnd)?;
 
         match tok.kind() {
-            TokenKind::Keyword(Keyword::Function) => {
+            TokenKind::Keyword(Keyword::Function) | TokenKind::Keyword(Keyword::Async) => {
```

Comment:
```suggestion
            TokenKind::Keyword(Keyword::Function | Keyword::Async) => {
```

Reply:
Ah! it seems I'm on an outdated rust version

Reply:
Actually, it seems this is still experimental as of 1.47.0?

Reply:
> Actually, it seems this is still experimental as of 1.47.0?

ah sorry about that, I thought it was stable